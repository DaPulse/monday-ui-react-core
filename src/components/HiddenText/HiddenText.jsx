import cx from "classnames";
import React, { forwardRef, useRef } from "react";
import PropTypes from "prop-types";
import useMergeRefs from "../../hooks/useMergeRefs";
import { ELEMENT_TYPES, getTestId } from "../../utils/test-utils";
import styles from "./HiddenText.module.scss";

const HiddenText = forwardRef(({ className, text, id, "data-testId": dataTestId }, ref) => {
  const componentRef = useRef(null);
  const mergedRef = useMergeRefs({ refs: [ref, componentRef] });

  return (
    <span
      ref={mergedRef}
      id={id}
      data-testid={dataTestId || getTestId(ELEMENT_TYPES.HIDDEN_TEXT, id)}
      className={cx(styles.wrapper, "hidden-text--wrapper", className)}
    >
      {text}
    </span>
  );
});

HiddenText.propTypes = {
  /** For overriding default styles. */
  className: PropTypes.string,
  /** So that it can be used for aria-labelledby values. */
  id: PropTypes.string,
  text: PropTypes.string.isRequired
};
HiddenText.defaultProps = {
  className: "",
  id: "hiddenText"
};

export default HiddenText;
