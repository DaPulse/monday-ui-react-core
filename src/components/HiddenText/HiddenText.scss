@import "../../styles/themes.scss";
@import "../../styles/typography.scss";

.hidden-text--wrapper {
  display: none;
}
