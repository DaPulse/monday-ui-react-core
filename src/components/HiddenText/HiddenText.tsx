import React, { forwardRef, useRef } from "react";
import cx from "classnames";
import useMergeRefs from "../../hooks/useMergeRefs";
import { ELEMENT_TYPES, getTestId } from "../../utils/test-utils";
import VibeComponentProps from "src/types/VibeComponentProps";
import styles from "./HiddenText.module.scss";

interface HiddenTextProps extends VibeComponentProps {
  text: string;
}

const HiddenText = forwardRef<HTMLSpanElement, HiddenTextProps>(
  ({ text, className = "", id = "hiddenText", "data-testid": dataTestId }, ref) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });

    return (
      <span
        ref={mergedRef}
        id={id}
        data-testid={dataTestId || getTestId(ELEMENT_TYPES.HIDDEN_TEXT, id)}
        className={cx(styles.wrapper, "hidden-text--wrapper", className)}
      >
        {text}
      </span>
    );
  }
);

export default HiddenText;
