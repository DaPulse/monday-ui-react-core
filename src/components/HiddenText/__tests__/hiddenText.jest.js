import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import HiddenText from "../HiddenText";

it("renders correctly with empty props", () => {
  const tree = renderer.create(<HiddenText />).toJSON();
  expect(tree).toMatchSnapshot();
});

// This test currently fails due to an apparent jest bug.
// it("Doesn't show the text", () => {
//   const { getByText } = render(<HiddenText text="Example" />);
//   const element = getByText("Example");
//   expect(element).not.toBeVisible();
// });
