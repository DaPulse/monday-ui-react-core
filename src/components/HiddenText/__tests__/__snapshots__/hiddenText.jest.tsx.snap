// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with empty props 1`] = `
<span
  className="wrapper hidden-text--wrapper"
  data-testid="hidden-text_hiddenText"
  id="hiddenText"
>
  This text is hidden
</span>
`;
